.github/workflows/build-mixxx.yml

```diff
@@ -0,0 +1,110 @@
+name: build Mixxx
+
+on: [pull_request, push]
+
+env:
+  #TODO: Set -DDEBUG_ASSERTIONS_FATAL=OFF before deploying CI builds as releases!!!
+  CMAKEFLAGS: -DCMAKE_BUILD_TYPE=Release -DBATTERY=ON -DBROADCAST=ON -DBULK=ON -DDEBUG_ASSERTIONS_FATAL=ON -DHID=ON -DLILV=ON -DOPUS=ON -DQTKEYCHAIN=ON -DVINYLCONTROL=ON
+  # GitHub Actions automatically zstd compresses caches
+  CCACHE_NOCOMPRESS: true
+
+jobs:
+  build-mac:
+    name: macOS
+    runs-on: macos-10.15
+    env:
+      MIXXX_ENVNAME: 2.3-1db6c06-sdk10.15-macosminimum10.12-x86_64
```

Comment:
TODO parse this from the repo file

Reply:
Why?

Reply:
That's how we always did it until now (`build/osx/golden_environment`). It makes it  possible tp add helper scripts for each OS that fetch and set up the environment (CMAKE_PREFIX_PATH) for the current machine. That could make it easier for users to configure their mixxx build properly and we could reuse them here.

Reply:
I suppose it does make it easier to look back in the Git history when the build environment was updated. Changing that old file will break Travis builds (and Jenkins, but that's already disabled). Shall I go ahead and disable Travis in this PR?

Reply:
I will hack around this by making a new file in the `cmake` directory instead of the old `build/osx`

Reply:
Nevermind. I do not think we should do this. The build environment name is [used for a cache key](https://github.com/mixxxdj/mixxx/pull/3305/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R128). I don't know how we could do that if it had to be read from a file.

Reply:
It's possible. Set `id: setup_buildenv` of the step and print this to the terminal during `run`. 

    ::set-output name=buildenv_name::my-custom-buildenv-name

In a later step, you can then do:

    key: buildenv-cache-${{ steps.setup_buildenv.outputs.buildenv_name }}

Reply:
Anyway, no need to do it now, just leave a TODO comment.

Reply:
That seems overcomplicated for very little benefit.

Reply:
Making the the dev setup easier for contributors is definitely worth 3 additional lines in a CI config file IMHO.

Reply:
Okay, done. I also got the macOS minimum version number from the build environment name so this CI script doesn't need to be updated when that changes.